sklearn/ensemble/_hist_gradient_boosting/tests/test_grower.py

```diff
@@ -383,5 +385,5 @@ def test_split_on_nan_with_infinite_values():
     predictions = predictor.predict(X)
     predictions_binned = predictor.predict_binned(
         X_binned, missing_values_bin_idx=bin_mapper.missing_values_bin_idx_)
-    assert np.all(predictions == -gradients)
-    assert np.all(predictions_binned == -gradients)
+    assert np.all(predictions.astype(G_H_DTYPE) == -gradients)
+    assert np.all(predictions_binned.astype(G_H_DTYPE) == -gradients)
```

Comment:
I'm a bit puzzled too.

The computation of the leaf values now happens in Cython, whereas before it was happening in Python.

I think it has to do with numpy refusing to compare 2 arrays with different dtypes, which is still weird because it's usually OK with that

I wouldn't worry too much about it though because the values are still definitely the same, and there is no precision loss.